dags/elsevier/metadata_parser.py

```diff
@@ -0,0 +1,113 @@
+import os
+from datetime import datetime
+
+from common.parsing.parser import IParser
+from common.parsing.xml_extractors import CustomExtractor
+from common.utils import extract_text, get_license_type_and_version_from_url
+from structlog import get_logger
+
+
+class ElsevierMetadataParser(IParser):
+    def __init__(self, doi, file_path) -> None:
+        self.dois = doi
+        self.file_path = file_path
+        self.year = None
+        self.journal_doctype = None
+        self.collaborations = []
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+        self.extractors = [
+            CustomExtractor(
+                destination="journal_title",
+                extraction_function=self._get_journal_title,
+                required=True,
+            ),
+            CustomExtractor(
+                destination="date_published",
+                extraction_function=self._get_published_date,
+                required=True,
+            ),
+            CustomExtractor(
+                destination="journal_year",
+                extraction_function=self._get_journal_year,
+                required=True,
+            ),
+            CustomExtractor(
+                destination="collections",
+                extraction_function=self._get_collections,
+            ),
+            CustomExtractor(
+                destination="license",
+                extraction_function=self._get_license,
+                required=True,
+            ),
+            # CustomExtractor(
+            #     destination="local_files",
+            #     extraction_function=self._get_local_files,
+            #     # required=True,
+            # ),
+        ]
+
+    def parse(self, article):
+        extracted_value = {}
+        journal_issues = article.findall("dataset-content/journal-item")
+        for journal_issue in journal_issues:
+            doi = journal_issue.find("journal-item-unique-ids/doi")
+            if doi.text == self.dois:
```

Comment:
because there is more than one article's metadata in the same file, and they have their own block with data.
For example:
```
			<journal-item-unique-ids>
				<pii>S0370-2693(23)00355-6</pii>
				<doi>10.1016/j.physletb.2023.138021</doi>
				<jid-aid>
					<jid>PLB</jid>
					<issn>0370-2693</issn>
					<aid>138021</aid>
					<article-number>138021</article-number>
				</jid-aid>
			</journal-item-unique-ids>
		.......
			<journal-item-unique-ids>
				<pii>S0370-2693(23)00378-7</pii>
				<doi>10.1016/j.physletb.2023.138044</doi>
				<jid-aid>
					<jid>PLB</jid>
					<issn>0370-2693</issn>
					<aid>138044</aid>
					<article-number>138044</article-number>
				</jid-aid>
			</journal-item-unique-ids>
```

Reply:
ok 👍